libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/tripsession/TripSessionComponent.kt

```diff
@@ -0,0 +1,96 @@
+package com.mapbox.navigation.dropin.component.tripsession
+
+import android.annotation.SuppressLint
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.repeatOnLifecycle
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.trip.session.TripSessionState
+import com.mapbox.navigation.ui.app.internal.Store
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
+import com.mapbox.navigation.ui.app.internal.tripsession.TripSessionStarterState
+import com.mapbox.navigation.ui.base.lifecycle.UIComponent
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.launch
+
+/**
+ * Component class that is responsible for starting and stopping [MapboxNavigation] trip session
+ * and replay trip session when [TripSessionStarterState] changes.
```

Comment:
What's the reason for moving this out of the `TripSessionStarterStateController`?

Similar to `AudioGuidanceStateController`, the reason the logic lived in a state controller is so that it can be shared with android auto.

If it lives in a component, then android auto will need to add it to the `MainCarSession` and I think there will be race conditions because there can be 2 `TripSessionComponent` trying to control the trip session state.

Reply:
Can this be a different PR? One of the next steps with the TripSessionStarterStateController, is to use it with Android Auto. I'm concerned that this change will make that harder

Reply:
Agreed, I remembered we had this discussion earlier and we decided not to host `TripSessionComponent` to avoid possible race conditions with android auto

Reply:
> We must start or stop the trip session on NavigationView lifecycle `Lifecycle.State.STARTED`.

Why is this required? We start the trip session when Drop-In UI is instantiated and permissions are granted. `mapboxNavigation` stops trip session automatically when `end navigation` on the notification is clicked. Trying to wrap my head around as to why `We must start or stop` based on lifecycle changes?

Reply:
> Why is this required?

I answered this in a comment below https://github.com/mapbox/mapbox-navigation-android/pull/6039#issuecomment-1183792520

Reply:
Short answer is `TripSessionComponent` is added to host the `Lifecycle` dependency which is not available in a `TripSessionStarterStateController`. Is that right?